packages/core/admin/admin/src/components/MainNav/NavLink.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+
+import { Tooltip, Icon, Flex, Badge } from '@strapi/design-system';
+import { NavLink as RouterLink, LinkProps } from 'react-router-dom';
+import styled from 'styled-components';
+
+export interface NavLinkProps extends LinkProps {
+  children: React.ReactNode;
+  icon?: React.ElementType<React.SVGProps<SVGSVGElement>>;
+  badgeAriaLabel?: string;
+  badgeContent?: string | number;
+}
+
+const MainNavLinkWrapper = styled(RouterLink)`
+  text-decoration: none;
+  display: block;
+  border-radius: ${({ theme }) => theme.borderRadius};
+  background: ${({ theme }) => theme.colors.neutral0};
+  color: ${({ theme }) => theme.colors.neutral600};
+  position: relative;
+
+  svg path {
+    fill: ${({ theme }) => theme.colors.neutral500};
+  }
+
+  &:hover,
+  &.active {
+    background: ${({ theme }) => theme.colors.neutral100};
+  }
+
+  &:hover {
+    svg path {
+      fill: ${({ theme }) => theme.colors.neutral600};
+    }
+    color: ${({ theme }) => theme.colors.neutral700};
+  }
+
+  &.active {
+    svg path {
+      fill: ${({ theme }) => theme.colors.primary600};
+    }
+
+    color: ${({ theme }) => theme.colors.primary600};
+    font-weight: 500;
+  }
+`;
+
+const CustomBadge = styled(Badge)`
+  span {
+    color: ${({ theme }) => theme.colors.neutral0};
+    line-height: 0;
+  }
```

Comment:
can you add notes why you have to do this, it's not clear & it might be something we can improve in the DS :)

Reply:
I need just to overwrite the border radius because in the Badge content there is already a border radius set for the Wrapper https://github.com/strapi/design-system/blob/958fe5a991ad51dc80ed3ce93f2b8f8d5763e451/packages/strapi-design-system/src/Badge/Badge.tsx#L7